frontend/components/NavBar.js

```diff
@@ -195,17 +195,21 @@ function LoginButton() {
 }
 
 function LogoutButton() {
+  const router = useRouter()
+
   const onClick = async () => {
     const endpoint = `${process.env.KITSPACE_GITEA_URL}/user/kitspace/sign_out`
-    const response = await fetch(endpoint, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-      },
-      body: `_csrf=${window.session._csrf}&`,
-      credentials: 'include',
-    })
-    console.log(response)
+    const xhr = new XMLHttpRequest()
+
+    xhr.open('POST', endpoint, true)
+    xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded')
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+        router.push('/')
+      }
+    }
+
+    xhr.send(`_csrf=${window.session._csrf}`)
```

Comment:
Why are you switching to XHR!?

Reply:
From the commit message:

- The sign out mechanism expects a request done with url-encoded-form, also using xhr not fetch.
- The headers get set correctly though `i_like_gitea`, `_csrf`, and `lang` for some reason get set twice in the response.
- One of them is from the sign out endpoint which clear them(the headers).
- Haven't identified the second source yet. It sets a value to them which makes the sign out fails.

I have been debugging it for a while now not sure if `fetch` is one of the reason, will try to get it to work with xhr then it will be easy to migrate to fetch.

Reply:
Calling the original logout endpoint solved the problem. I Will see if it can be done with fetch.